src/py/flwr/superexec/exec_servicer.py

```diff
@@ -53,13 +57,66 @@ def StartRun(
 
         self.runs[run.run_id] = run
 
+        # Start a background thread to capture the log output
+        capture_thread = threading.Thread(
+            target=self._capture_logs, args=(run,), daemon=True
+        )
+        capture_thread.start()
+
         return StartRunResponse(run_id=run.run_id)
 
-    def StreamLogs(
+    def _capture_logs(
```

Comment:
Can we make this a standalone function instead of a class method? I think this would improve testability.

Reply:
On that front - can we include the tests in this PR?

Reply:
Sure! This makes sense especially if we remove `self.lock` and set `self.select_timeout` as a constant. I'll update this.